cpp/src/arrow/type_traits.h

```diff
@@ -292,7 +291,16 @@ struct TypeTraits<ExtensionType> {
 //
 
 template <typename T>
-using is_number = std::is_base_of<Number, T>;
```

Comment:
If we're renaming this one, should we do the same for `is_signed_integer` etc?

Reply:
Probably. There's some other cleanup to do but not here, I will open a JIRA